### ISSUE TITLE
`dvc queue run <hash>` to run a specific experiment

### ISSUE THREAD

**Issue by PythonFZ** [iterative/dvc#8121]
I'm using the dvc experiment feature together with the queue to plan multiple experiments I want to run.
I wanted to run an already queued experiment to check its outputs without submitting all jobs.

My Idea would be something like
`dvc queue run <hash>` which will run the given experiment in `.dvc/tmp/` and upon success remove it from the queue.

**Comment by dberenbaum** [comment]
Keeping this feature request open, but in case you want a workaround for this in the meantime, you can do:

```
$ dvc exp apply <queue_id>
$ dvc exp run --temp
```

**Comment by PythonFZ** [comment]
Thanks, I had to add `dvc queue remove` to clean the queue from the old experiment which would just be a copy. But this was more or less exactly what I was looking for.

```sh
dvc exp apply $queue_id
dvc exp run --temp
dvc queue remove $queue_id
```

**Comment by dberenbaum** [comment]
Maybe `dvc queue pop <hash>` would make sense as an alias for that?

**Comment by PythonFZ** [comment]
It probably can't be `dvc queue run <hash>` because the id changes so I like the `dvc queue pop <hash>` idea.
How would this affect the workspace? If you use `apply` + `run --temp` it will change the workspace, so maybe use stash the workflow before the `apply` and then apply the stash afterwards?

**Comment by PythonFZ** [comment]
In addition to `dvc queue pop <id>` one could also have just `dvc queue pop` to start the next experiment in line. In this way one can run multiple experiments in temporary directories from the same repository. This could be especially useful on shared file systems to run experiment from different machines.

**Comment by PythonFZ** [comment]
I found a solution for myself by writing https://github.com/zincware/dask4dvc and adding this feature via `dask4dvc run <id>` which will reproduce the specific experiment.

**Comment by gregstarr** [comment]
> Thanks, I had to add `dvc queue remove` to clean the queue from the old experiment which would just be a copy. But this was more or less exactly what I was looking for.
> 
> ```shell
> dvc exp apply $queue_id
> dvc exp run --temp
> dvc queue remove $queue_id
> ```

does this work in the case of multiple workers assuming the workers aren't trying to do `dvc exp apply` at the same time? once the call to `dvc exp run --temp` is called, can the workspace be safely changed by another worker with `dvc exp apply`?

**Comment by dberenbaum** [comment]
@gregstarr Could you clarify your concern or explain further?

**Comment by gregstarr** [comment]
I'll use tmux for an example:

Assume I already have a list of experiments queued. I open a window, cd to my repo and begin the experiment using `dvc exp apply $queue_id`, `dvc exp run --temp`. Can I then open another tmux window and do the same steps again for a different experiment?

**Comment by dberenbaum** [comment]
Yes, as long as you are okay that each `dvc exp apply` call will change the state of your workspace. `dvc exp run --temp` will create a new isolated workspace for each one and run them in parallel.

**Comment by gregstarr** [comment]
Okay, is there a command which would restore the workspace to how it was before `dvc exp apply` was called?

**Comment by PythonFZ** [comment]
You can `git stash --include-untracked` before you `dvc exp apply`and see if `git stash pop` does work.

**Comment by gregstarr** [comment]
how does dvc deal with this issue? when I queue up experiments with `dvc exp run --queue -S param1=val1 ...` the workspace is left unchanged afterwards right?

**Comment by dberenbaum** [comment]
@gregstarr PTAL at https://dvc.org/doc/command-reference/exp/apply#expand-for-details-on-reverting-object-object and see if that helps

**Comment by dberenbaum** [comment]
Correct. @pmrowla please correct if you see anything wrong above.

**Comment by gregstarr** [comment]
thanks 👍
